src/generator/instr_generator.rs

```diff
@@ -22,28 +23,27 @@ fn get_loc_info<'a>(rule: &'a WhammProvider, emitter: &VisitingEmitter) -> Optio
 /// passed emitter to emit instrumentation code.
 /// This process should ideally be generic, made to perform a specific
 /// instrumentation technique by the passed Emitter type.
-pub struct InstrGenerator<'a, 'b, 'c, 'd, 'e> {
-    pub emitter: VisitingEmitter<'a, 'b, 'c, 'd>,
+pub struct InstrGenerator<'a, 'b, 'c, 'd, 'e, 'f, 'g> {
+    pub emitter: VisitingEmitter<'a, 'b, 'c, 'd, 'e, 'f>,
     pub ast: SimpleAST,
-    pub err: &'e mut ErrorGen,
-
+    pub err: &'g mut ErrorGen,
     curr_instr_args: Vec<Arg>,
     curr_probe_mode: String,
     /// The current probe's body and predicate
     curr_probe: Option<(Option<Vec<Statement>>, Option<Expr>)>,
 }
-impl<'a, 'b, 'c, 'd, 'e> InstrGenerator<'a, 'b, 'c, 'd, 'e> {
+impl<'a, 'b, 'c, 'd, 'e, 'f, 'g> InstrGenerator<'a, 'b, 'c, 'd, 'e, 'f, 'g> {
     pub fn new(
-        emitter: VisitingEmitter<'a, 'b, 'c, 'd>,
+        emitter: VisitingEmitter<'a, 'b, 'c, 'd, 'e, 'f>,
         ast: SimpleAST,
-        err: &'e mut ErrorGen,
+        err: &'g mut ErrorGen,
     ) -> Self {
         Self {
             emitter,
             ast,
             err,
             curr_instr_args: vec![],
-            curr_probe_mode: "".to_string(),
+            curr_probe_mode: "____0".to_string(),
```

Comment:
What's this change for?

Reply:
to have current location information in InstrGenerator that can be collected, then passed all at once to the report_var_metadata to be made into 1 "metadata" object for inserts